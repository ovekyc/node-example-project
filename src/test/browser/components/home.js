import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import Home from '../../../front/components/home';

it('snapshot test', ()=> {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('<Home />', () => {
  it('should render without throwing an error', () => {
    expect(shallow(<Home />)
      .contains(
        <div className="home">
          <h1>Hello world!</h1>
        </div>))
      .toBe(true);
  });

  it('should be selectable by class "home"', () => {
    expect(shallow(<Home />)
      .is('.home'))
      .toBe(true);
  });

  it('should mount in a full DOM', function() {
    expect(mount(<Home />).find('.home').length)
      .toBe(1);
  });

  it('should render to static HTML', function() {
    expect(render(<Home />).text())
      .toEqual('Hello world!');
  });
});
