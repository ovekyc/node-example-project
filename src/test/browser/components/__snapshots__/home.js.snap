// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="home"
>
  <h1>
    Hello world!
  </h1>
</div>
`;
